### ISSUE TITLE
feat: add application support for Linux

### ISSUE THREAD

**Issue by dmiedev** [VeryGoodOpenSource/very_good_cli#452]
It would be nice to have out-of-the-box support for macOS and Linux. Right now an app created with `very_good_cli` only supports Android, iOS, Windows and Web.

Sorry if this is a duplicate, I couldn't find a similar issue here.

**Comment by felangel** [comment]
Hi @dmiedev 👋 
Thanks for opening an issue!

Unfortunately we are still blocked on https://github.com/flutter/flutter/issues/64088 for MacOS. I believe we should be able to add Linux support though -- I'll verify shortly 👍

**Comment by see** [comment]
same issue.

**Comment by oravecz** [comment]
Is Flavors still the best approach for platform-specific configuration in Flutter? I don't want to spread misinformation, but Flavors was borrowed from Android where it is well-supported, but the Flutter dev docs have precious little information regarding the use of flavors cross-platform and delegate to a bunch of articles which talk about how it isn't well-supported in iOS (or MacOS obv). Having a separate entry point per flavor might seem ok (does it?) when we are talking about dev/test/prod, but what about when the permutations start to grow to include a/b testing and canary/staging/etc?

There are other libraries for sharing configuration between native and Dart. Why not use these instead?
* https://pub.dev/packages/flutter_config
* https://pub.dev/packages/environment_config

caveat - I have not used these, but plan to evaluate

**Comment by deveraman** [comment]
Hey @felangel could we expect the support for macOS since [118421]https://github.com/flutter/flutter/pull/118421 was merged recently?

Also Linux support would be great as well.

**Comment by felangel** [comment]
> Hey @felangel could we expect the support for macOS since [#118421](https://github.com/flutter/flutter/pull/118421) was merged recently?
> 
> Also Linux support would be great as well.

Hey 👋 
Yes we’re planning to work on this very soon. Flavor support in MacOS is not yet available on the stable channel afaik but we plan to address this soon after the changes are available on stable.

**Comment by josephybh** [comment]
@felangel Any update?

**Comment by rasmus-dybmose** [comment]
Any update?

**Comment by egyleader** [comment]
me also voting for this issue as macos flavoring updated in flutter 3.10

**Comment by renancaraujo** [comment]
> macos flavoring updated in flutter 3.10

This is an excellent issue for people willing to contribute!

We have this in our backlog, but this is not at the top of our priorities compared to other stuff we have to do. If you are reading this and think you can do it, we have a great [contributing guide](https://github.com/VeryGoodOpenSource/very_good_cli/blob/main/CONTRIBUTING.md) that may help to find where to plug the wires.

This should be done on the following templates:
- [Very good core](https://github.com/VeryGoodOpenSource/very_good_core) (start app)
- [Very good wear app](https://github.com/VeryGoodOpenSource/very_good_wear_app)
- [Very good flame game](https://github.com/VeryGoodOpenSource/very_good_flame_game)

**Comment by mxknt** [comment]
I'd like to try and take this on, currently working on adding macos support to the starter app template!

**Comment by mxknt** [comment]
PR is up for the starter app template! I forgot this was for both macOS and Linux though - I don't have access to a Linux machine for development so I would only be able to do the macOS side of this.

**Comment by renancaraujo** [comment]
Thanks @mxknt so much for your contributions. Renaming this issue to track the remaining linux platform.

**Comment by mxknt** [comment]
Sure thing!
